subprojects/common/core/src/main/java/hu/bme/mit/theta/core/utils/IEEEBigDecimal.java

```diff
@@ -0,0 +1,346 @@
+			} else {
+				assert bigDecimal != null;
+				return bigDecimal.compareTo(ieeeBigDecimal.bigDecimal) < 0;
+			}
+		}
+	}
+
+	public boolean isLessThanEqual(final IEEEBigDecimal ieeeBigDecimal) {
+		if (this == NaN || ieeeBigDecimal == NaN) {
+			return false;
+		}
+		if (this == POSITIVE_INFINITY) {
+			return ieeeBigDecimal == POSITIVE_INFINITY;
+		} else if (this == NEGATIVE_INFINITY) {
+			return true;
+		} else {
+			if (ieeeBigDecimal == POSITIVE_INFINITY) {
+				return true;
+			} else if (ieeeBigDecimal == NEGATIVE_INFINITY) {
+				return false;
+			} else {
+				assert bigDecimal != null;
+				return bigDecimal.compareTo(ieeeBigDecimal.bigDecimal) <= 0;
+			}
+		}
+	}
+
+	public boolean isGreaterThan(final IEEEBigDecimal ieeeBigDecimal) {
+		if (this == NaN || ieeeBigDecimal == NaN) {
+			return false;
+		}
+		if (this == POSITIVE_INFINITY) {
+			return ieeeBigDecimal != POSITIVE_INFINITY;
+		} else if (this == NEGATIVE_INFINITY) {
+			return false;
+		} else {
+			if (ieeeBigDecimal == POSITIVE_INFINITY) {
+				return false;
+			} else if (ieeeBigDecimal == NEGATIVE_INFINITY) {
+				return true;
+			} else {
+				assert bigDecimal != null;
+				return bigDecimal.compareTo(ieeeBigDecimal.bigDecimal) > 0;
+			}
+		}
+	}
+
+	public boolean isGreaterThanEqual(final IEEEBigDecimal ieeeBigDecimal) {
+		if (this == NaN || ieeeBigDecimal == NaN) {
+			return false;
+		}
+		if (this == POSITIVE_INFINITY) {
+			return true;
+		} else if (this == NEGATIVE_INFINITY) {
+			return ieeeBigDecimal == NEGATIVE_INFINITY;
+		} else {
+			if (ieeeBigDecimal == POSITIVE_INFINITY) {
+				return false;
+			} else if (ieeeBigDecimal == NEGATIVE_INFINITY) {
+				return true;
+			} else {
+				assert bigDecimal != null;
+				return bigDecimal.compareTo(ieeeBigDecimal.bigDecimal) >= 0;
+			}
+		}
+	}
+
+	public BigInteger unscaledValue() {
+		if (this == NaN || this == POSITIVE_INFINITY || this == NEGATIVE_INFINITY) {
+			throw new IllegalStateException("NaN or Infinity does not have unscaled value");
+		}
+		assert bigDecimal != null;
+		return bigDecimal.unscaledValue();
+	}
+
+	public int scale() {
+		if (this == NaN || this == POSITIVE_INFINITY || this == NEGATIVE_INFINITY) {
+			throw new IllegalStateException("NaN or Infinity does not have scale");
+		}
+		assert bigDecimal != null;
+		return bigDecimal.scale();
+	}
+
+	@Override
+	public boolean equals(final Object x) {
+		if (this == NaN || x == NaN) return this == x;
+		else if (this == POSITIVE_INFINITY || x == POSITIVE_INFINITY) return this == x;
+		else if (this == NEGATIVE_INFINITY || x == NEGATIVE_INFINITY) return this == x;
+		else {
+			assert bigDecimal != null;
+			return bigDecimal.equals(x);
+		}
+	}
+
+	@Override
+	public String toString() {
+		if (this == NaN) return "NaN";
+		else if (this == POSITIVE_INFINITY) return "+Infinity";
+		else if (this == NEGATIVE_INFINITY) return "-Infinity";
+		else {
+			assert bigDecimal != null;
+			return bigDecimal.toString();
+		}
+	}
+}
```

Comment:
I am not sure this file is used at all. As I recall, this was an early attempt at implementing the fp standard in java. Without much success I might add. I think it can be deleted.
```suggestion
```